src/nanotron/parallel/tensor_parallel/functional.py

```diff
@@ -338,21 +337,102 @@ def backward(ctx, grad_output):
             raise ValueError(f"Got unexpected mode: {tp_mode}.")
 
 
+class _ColumnLinearContextParallelNoAsync(torch.autograd.Function):
```

Comment:
Rename it _ColumnLinearNoAsyncCommunication to make it consistent with `_ColumnLinearAsyncCommunication`

Reply:
That class only supports reduce scatter. Should I make that distinction in the name? For instance, `_ColumnLinearAsyncCommunicationReduceScatterMode`? A bit too long haha

Reply:
yeah `_ColumnLinearNoAsyncCommunicationReduceScatterMode` for now is good. We will refactor later in another PR to make it consistent with other mode if needed